server/src/tree-sitter/analyzer.ts

```diff
@@ -1129,23 +1137,27 @@ export default class Analyzer {
   }
 
   /**
-   *
-   * @param scanResult
-   * @param uri
-   *
    * Process scan results sent from the client. The scan results are generated by bitbake -e <recipe> command
    */
   public processRecipeScanResults (scanResult: string, chosenRecipe: string): void {
-    if (this.bitBakeParser === undefined) {
-      logger.debug('[ProcessRecipeScanResults] The analyzer is not initialized with a parser')
-      return undefined
-    }
-
     if (chosenRecipe === undefined) {
       logger.error('[ProcessRecipeScanResults] The chosenRecipe is undefined, abort processing scan results')
       return
     }
 
+    this.uriToLastScanResult[chosenRecipe] = this.processEnvScanResults(scanResult)
+  }
+
+  public processGlobalEnvScanResults (scanResult: string): void {
+    this.lastGlobalEnvScanResult = this.processEnvScanResults(scanResult)
+  }
+
+  private processEnvScanResults (scanResult: string): LastScanResult | undefined {
+    if (this.bitBakeParser === undefined) {
+      logger.debug('[ProcessRecipeScanResults] The analyzer is not initialized with a parser')
```

Comment:
```suggestion
      logger.debug('[processEnvScanResults] The analyzer is not initialized with a parser')
```

Reply:
Cool, you have sharp eyes. Now it is done in a non-hard-coded way.